gulpfile.js

```diff
@@ -568,109 +576,25 @@ const testSkillmap = gulp.series(buildSkillmapTests, copySkillmapTests, runSkill
-    .pipe(gulp.dest(`${multiplayerOut}/js`));
-
-const copyMultiplayerHtml = () => rimraf("webapp/public/multiplayer.html")
-    .then(() => gulp.src(`${multiplayerRoot}/build/index.html`)
-                    .pipe(replace(/="\/static\//g, `="/blb/multiplayer/`))
-                    .pipe(concat("multiplayer.html"))
-                    .pipe(gulp.dest("webapp/public")));
-
-const multiplayer = gulp.series(cleanMultiplayer, buildMultiplayer, gulp.series(copyMultiplayerCss, copyMultiplayerJs, copyMultiplayerHtml));
+const multiplayer = createWebappTasks("multiplayer");
 
 /********************************************************
                       Kiosk
 *********************************************************/
 
-const kioskRoot = "kiosk";
-const kioskOut = "built/web/kiosk";
-
-const cleanKiosk = () => rimraf(kioskOut);
-
-const npmBuildKiosk = () => exec("npm run build", true, { cwd: kioskRoot });
-
-const buildKiosk = async () => await npmBuildKiosk();
-
-const copyKioskCss = () => gulp.src(`${kioskRoot}/build/static/css/*`)
-    .pipe(gulp.dest(`${kioskOut}/css`));
-
-const copyKioskJs = () => gulp.src(`${kioskRoot}/build/static/js/*`)
-    .pipe(gulp.dest(`${kioskOut}/js`));
-
-const copyKioskHtml = () => rimraf("webapp/public/kiosk.html")
-    .then(() => gulp.src(`${kioskRoot}/build/index.html`)
-                    .pipe(replace(/="\/static\//g, `="/blb/kiosk/`))
-                    .pipe(concat("kiosk.html"))
-                    .pipe(gulp.dest("webapp/public")));
-
-const kiosk = gulp.series(cleanKiosk, buildKiosk, gulp.series(copyKioskCss, copyKioskJs, copyKioskHtml));
+const kiosk = createWebappTasks("kiosk");
 
 /********************************************************
                       Teacher Tool
 *********************************************************/
 
-const teacherToolRoot = "teachertool";
-const teacherToolOut = "built/web/teachertool";
-
-const cleanTeacherTool = () => rimraf(teacherToolOut);
-
-const npmBuildTeacherTool = () => exec("npm run build", true, { cwd: teacherToolRoot });
-
-const buildTeacherTool = async () => await npmBuildTeacherTool();
-
-const copyTeacherToolCss = () => gulp.src(`${teacherToolRoot}/build/static/css/*`)
-    .pipe(gulp.dest(`${teacherToolOut}/css`));
-
-const copyTeacherToolJs = () => gulp.src(`${teacherToolRoot}/build/static/js/*`)
-    .pipe(gulp.dest(`${teacherToolOut}/js`));
-
-const copyTeacherToolHtml = () => rimraf("webapp/public/teachertool.html")
-    .then(() => gulp.src(`${teacherToolRoot}/build/index.html`)
-                    .pipe(replace(/="\/static\//g, `="/blb/teachertool/`))
-                    .pipe(concat("teachertool.html"))
-                    .pipe(gulp.dest("webapp/public")));
-
-const teacherTool = gulp.series(cleanTeacherTool, buildTeacherTool, gulp.series(copyTeacherToolCss, copyTeacherToolJs, copyTeacherToolHtml));
+const teacherTool = createWebappTasks("teachertool");
```

Comment:
Down below here, we have the "webapp build wrapper" section which calls into the update*Strings functions. That could maybe be made slightly simpler with a unified updateWebappStrings function so we don't have to create a new one for each app, but not a huge deal.

Reply:
done!